lib/nerves_hub_web/live/devices/device_health.ex

```diff
@@ -160,67 +164,57 @@ defmodule NervesHubWeb.Live.Devices.DeviceHealth do
     Enum.map(data, &Map.get(&1, :type))
   end
 
-  def create_chart_data(device_id, time_frame, memory_size) do
-    default = create_default_chart_data(device_id, time_frame, memory_size)
-    custom = create_custom_chart_data(device_id, time_frame)
-
-    # Concat default and custom metrics and keep only non-nil results
-    Enum.concat(default, custom)
-    |> Enum.filter(& &1)
-  end
-
-  def create_default_chart_data(device_id, {unit, _} = time_frame, memory_size) do
-    @default_metric_types
-    |> Enum.map(fn {type, title} ->
-      data =
-        device_id
-        |> Metrics.get_device_metrics_by_key(Atom.to_string(type), time_frame)
-        |> get_max_per_hour(unit)
-        |> organize_metrics_for_chart()
-
-      unless data == [] do
-        %{
-          type: Atom.to_string(type),
-          title: title,
-          data: data,
-          max: get_max_value(type, data, memory_size),
-          min: 0,
-          unit: get_time_unit(time_frame)
-        }
-      end
-    end)
-  end
-
-  def create_custom_chart_data(device_id, {unit, _} = time_frame) do
+  def create_chart_data(device_id, {unit, _} = time_frame, memory_size) do
     device_id
-    |> Metrics.get_custom_metrics_for_device(time_frame)
+    |> Metrics.get_device_metrics(time_frame)
     |> Enum.group_by(& &1.key)
+    |> filter_and_sort()
     |> Enum.map(fn {type, metrics} ->
-      data =
-        metrics
-        |> get_max_per_hour(unit)
-        |> organize_metrics_for_chart()
-
-      title = String.replace(type, "_", " ") |> String.capitalize()
+      data = organize_metrics_for_chart(metrics, unit)
 
+      # Build structure for rendering charts
       %{
         type: type,
-        title: title,
+        title: title(type),
         data: data,
-        max: get_max_value(:custom, data),
+        max: get_max_value(type, data, memory_size),
         min: get_min_value(data),
         unit: get_time_unit(time_frame)
       }
     end)
   end
 
-  defp organize_metrics_for_chart(metrics) do
+  defp filter_and_sort(metrics) do
+    metrics
+    |> Enum.reject(fn {type, _} -> type in @no_chart_metrics end)
+    |> Enum.sort_by(fn {type, _} ->
+      # Sorts list by @default_metrics order
+      Enum.find_index(@default_metrics, fn {type_, _} ->
+        type_ == type
+      end)
```

Comment:
This was a bit confusing for me, I think this naming reads easier. What do you think?

```suggestion
      Enum.find_index(@default_metrics, fn {default_type, _} ->
        type == type
      end)
```

Reply:
Agree!